hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.hadoop;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.net.URI;
+import java.security.KeyPair;
+import javax.ws.rs.core.UriBuilder;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FilterFileSystem;
+import org.apache.hadoop.fs.Path;
+
+/**
+ * A {@link FileSystem} that transparently encrypts and decrypts the streams of an underlying FileSystem and stores
+ * encapsulated/wrapped encryption keys alongside the encrypted data in the same FileSystem as the encrypted data (see
+ * {@link FileKeyStorageStrategy}). Unlike the {@link EncryptedFileSystem}, the ConfigurableEncryptedFileSystem is
+ * initialized using a {@link Configuration} and can be used as a standalone FileSystem (like {@code s3a://, hdfs://,
+ * etc.}).
+ */
+public final class ConfigurableEncryptedFileSystem extends FilterFileSystem {
+
+    private static final String SCHEME = "";
+    private static final String DEFAULT_ALGORITHM = "RSA";
+
+    public static final String PUBLIC_KEY_CONF = "fs.efs.key.public";
```

Comment:
what format is expected for these? should also say that these are keys into the given Configuration object